hooks/tk-multi-publish2/basic/publish_session_geometry.py

```diff
@@ -0,0 +1,514 @@
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        item.properties["project_root"] = project_root
+
+        # get the configured work file template
+        work_template = item.parent.properties["work_file_template"]
+        publish_template = self._get_template("Publish file Template", settings)
+
+        # get the current scene path and extract fields from it
+        # using the work template:
+        scene_path = os.path.abspath(cmds.file(query=True, sn=True))
+        #scene_file_name = os.path.basename(scene_path)
+        scene_file_name = publisher.util.get_publish_name(scene_path)
+        scene_file_name_root, scene_file_name_ext = os.path.splitext(scene_file_name)
+        fields = work_template.get_fields(scene_path) #{ "asset_root": project_root, "name": scene_file_name_root, "version": 0 }
+                
+        # create the publish path by applying the fields 
+        # with the publish template:
+        publish_path = publish_template.apply_fields(fields)
+        
+        # ensure the publish folder exists:
+        publish_folder = os.path.dirname(publish_path)
+        self.parent.ensure_folder_exists(publish_folder)
+
+        # get the path in a normalized state. no trailing separator,
+        # separators are appropriate for current os, no double separators,
+        # etc.
+        path = sgtk.util.ShotgunPath.normalize(publish_path)
+
+        # determine the publish path, version, type, and name
+        publish_info = self._get_publish_info(path, settings)
+
+        publish_name = publish_info["name"]
+
+        # see if there are any other publishes of this path with a status.
+        # Note the name, context, and path *must* match the values supplied to
+        # register_publish in the publish phase in order for this to return an
+        # accurate list of previous publishes of this file.
+        publishes = publisher.util.get_conflicting_publishes(
+            item.context,
+            path,
+            publish_name,
+            filters=["sg_status_list", "is_not", None]
+        )
+
+        if publishes:
+            conflict_info = (
+                "If you continue, these conflicting publishes will no longer "
+                "be available to other users via the loader:<br>"
+                "<pre>%s</pre>" % (pprint.pformat(publishes),)
+            )
+            self.logger.warn(
+                "Found %s conflicting publishes in Shotgun" %
+                (len(publishes),),
+                extra={
+                    "action_show_more_info": {
+                        "label": "Show Conflicts",
+                        "tooltip": "Show the conflicting publishes in Shotgun",
+                        "text": conflict_info
+                    }
+                }
+            )
+
+        # store the session path since this is guaranteed to run just before
+        # the publish itself
+        item.properties["publish_file_path"] = path
```

Comment:
`path` isn't the publish path. I think we can calculate the publish info above and store it in the properties like the next line.

Reply:
Fixed